src/DGtal/shapes/parametric/Lemniscate2D.ih

```diff
@@ -111,8 +110,11 @@ inline
 typename DGtal::Lemniscate2D<T>::RealPoint2D
 DGtal::Lemniscate2D<T>::x( double t ) const
 {
-  RealPoint2D c( myA * sin(t)/(1+pow(cos(t),2)),
-                 myB * (sin(t)*cos(t)) / (1+pow(cos(t),2)) );
+  const double cost = cos(t);
+  const double sint = sin(t);
+  const double cos2t = pow(cost,2);
+  RealPoint2D c( myA * sint / (1. + cos2t),
+                 myA * sint * cost / (1. + cos2t) );
```

Comment:
For `t = pi` you will have division by zero, if `t` can reach `pi` then I suggest make some checks to avoid this to happen.

Reply:
After some tests, a call to this method with PI throws any error bu produce a good result, almost equal to 0.
I'm not sure that this behaviour is common to all environment.
Do you think I should still add a test?

Reply:
I would suggest handling this situation on your side. If you write a unit test with pi then it will fail due to the fact we now have detection of floating-point arithmetic errors, e.g., division by zero. The value may be correct but I am not sure if this behavior is expected on all platforms.